### PR TITLE
Modified SMTPEmailService to use Jtwig

### PR DESCRIPTION
**Pull request by kirkeaton**
1. Modified SMTPEmailService to use Jtwig templating library
2. Added DAOResourceLoader to handle when templates are including in other templates
3. Removed code that was being used with old template implementation

### CODE REVIEW

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  @Override
+  public InputStream load(String s) {
+    EmailTemplate template = (EmailTemplate) dao_.find(s);
+    return new ByteArrayInputStream(template.getBody().getBytes(StandardCharsets.UTF_8));
+  }
+
+  @Override
+  public boolean exists(String s) {
+    return ( dao_.find(s) != null );
```
Comment (kirkeaton) [inline src/foam/nanos/notification/email/DAOResourceLoader.java:34]: @kgrgreer Should I be caching the templates into a Map in this class?

File: src/foam/nanos/notification/email/DAOResourceLoader.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  @Override
+  public Optional<Charset> getCharset(String s) {
+    return Optional.absent();
+  }
+
+  @Override
+  public InputStream load(String s) {
+    EmailTemplate template = (EmailTemplate) dao_.find(s);
+    return new ByteArrayInputStream(template.getBody().getBytes(StandardCharsets.UTF_8));
```
Comment (kgrgreer) [inline src/foam/nanos/notification/email/DAOResourceLoader.java]: Wouldn't it be more direct to just use StringBufferInputStream?

File: src/foam/nanos/notification/email/DAOResourceLoader.java (hunk 1 above)
Comment (kgrgreer) [reply src/foam/nanos/notification/email/DAOResourceLoader.java:34]: If StringBufferInputStream works, then the overhead is negligible, so I wouldn't bother.

File: src/foam/nanos/notification/email/DAOResourceLoader.java (hunk 2 above)
Comment (kirkeaton) [reply src/foam/nanos/notification/email/DAOResourceLoader.java]: StringBufferInputStream is deprecated and the Java docs say this in regards to it:
>This class does not properly convert characters into bytes.

**Pull request merged**